src/mdns/mdns.cpp

```diff
@@ -209,10 +211,23 @@ void Publisher::AddServiceRegistration(ServiceRegistrationPtr &&aServiceReg)
     mServiceRegistrations.emplace(MakeFullServiceName(aServiceReg->mName, aServiceReg->mType), std::move(aServiceReg));
 }
 
-void Publisher::RemoveServiceRegistration(const std::string &aName, const std::string &aType)
+void Publisher::RemoveServiceRegistration(const std::string &aName, const std::string &aType, otbrError aError)
 {
+    auto                   it = mServiceRegistrations.find(MakeFullServiceName(aName, aType));
+    ServiceRegistrationPtr serviceReg;
+
     otbrLogInfo("Removing service %s.%s", aName.c_str(), aType.c_str());
-    mServiceRegistrations.erase(MakeFullServiceName(aName, aType));
+    VerifyOrExit(it != mServiceRegistrations.end());
```

Comment:
log after it's found?

Reply:
The log logs whether this method is called.